controllers/complianceeventsapi/server.go

```diff
@@ -754,6 +754,26 @@ func getSingleComplianceEvent(db *sql.DB, w http.ResponseWriter,
 	}
 }
 
+// getPqErrKeyVals is a helper to add additional database error details to a log message. additionalKeyVals is provided
+// as a convenience so that the keys don't need to be explicitly set to interface{} types when using the
+// `getPqErrKeyVals(err, "key1", "val1")...“ syntax.
+func getPqErrKeyVals(err error, additionalKeyVals ...interface{}) []interface{} {
+	unwrappedErr := err
+
+	for unwrappedErr != nil {
+		if pqErr, ok := unwrappedErr.(*pq.Error); ok { //nolint: errorlint
+			return append(
+				[]interface{}{"dbMessage", pqErr.Message, "dbDetail", pqErr.Detail, "dbCode", pqErr.Code},
+				additionalKeyVals...,
+			)
+		}
+
+		unwrappedErr = errors.Unwrap(unwrappedErr)
```

Comment:
This seems like the kind of thing that errors.Is or errors.As would do... but I readily admit being confused by those often. Do neither of those work here?

Reply:
I think `errors.As` could work as a type conversion: [errors.As](https://pkg.go.dev/errors) "examines the tree of its first argument looking for an error that can be assigned to its second argument, which must be a pointer. If it succeeds, it performs the assignment and returns true."

But I mention it more out of curiosity than necessity.

Reply:
Oh yeah, you're right. I didn't catch that the second argument is a pointer to the type that gets set if the type matches:
https://github.com/open-cluster-management-io/governance-policy-propagator/pull/202